src/ray/common/cgroup/cgroup_utils.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Util functions to setup cgroup.
+
+#pragma once
+
+#include <memory>
+#include <string_view>
+#include <utility>
+
+#include "ray/common/cgroup/cgroup_context.h"
+
+namespace ray {
+
+class CgroupV2Setup {
```

Comment:
Add comment about this class holds a cgroup v2 created in ctor and puts the pid into that cgroup in ctor, and do the opposite ie remove pid from cgroup and destroys cgroup in dtor.

Q: in ctor we move-in the pid from a "prev cgroup" into this cgroup. in dtor, do we need to move it back to the "prev cgroup"? if so, do we need to record that pid's original cgroup somewhere?

Also add that the cgroup deletion in dtor may fail if the cgroup is not empty (holds any other procs). If this is the case we print a WARNING and the cgroup is leaked.

Reply:
see the other comment about "prev cgroup" or "default cgroup"

Reply:
> Add comment about this class holds a cgroup v2 created in ctor and puts the pid into that cgroup in ctor, and do the opposite ie remove pid from cgroup and destroys cgroup in dtor.

Sure, comment added.

Reply:
I think we can make the comment more clear. Instead of saying "sets up" and "cleans up", we can say:

```
On ctor, creates a cgroup v2 if not already exists. Then puts `ctx.pid` into this cgroup.
On dtor, puts `ctx.pid` into the default cgroup, and tries to remove this cgroup v2.
```

Reply:
Updated.